engine/opts.go

```diff
@@ -72,7 +73,10 @@ func (m ClientMetadata) ToGRPCMD() metadata.MD {
 	if err != nil {
 		panic(err)
 	}
-	return metadata.Pairs(clientMetadataMetaKey, string(b))
+	return metadata.Pairs(
+		clientMetadataMetaKey,
+		base64.StdEncoding.EncodeToString(b),
```

Comment:
LGTM but IIUC the same problem might apply to the other opt structs here (`LocalImportOpts` and `LocalExportOpts`, e.g. if a path name on the host being imported/exported is `/🐈/🎅/🤡`), so worth throwing in the same thing for them.

Reply:
Also probably worth extending the integ test added the other day w/ coverage for UTF8 chars in the labels

Reply:
Looking at `LocalImportOpts` now, I can see that we also have to set some Buildkit-compatible headers to raw values:

https://github.com/dagger/dagger/blob/30e48293b2a287aca3957482ebb71150fb975df4/engine/opts.go#L126-L130

I'll at least change it to use the same code + pattern for anywhere that we're `json.Marshal`ing, but it seems like there's still a timebomb here. Seems unlikely at least, and it would affect Buildkit too.